data/store/mongo/mongo_test.go

```diff
@@ -276,26 +283,29 @@ var _ = Describe("Mongo", Label("mongodb", "slow", "integration"), func() {
 		var dataSetCollection *mongo.Collection
 		var summaryCollection *mongo.Collection
 		var alertsCollection *mongo.Collection
+		var collectionsOnce sync.Once
 
 		BeforeEach(func() {
-			var err error
-			store, err = dataStoreMongo.NewStore(config)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(store).ToNot(BeNil())
-			collection = store.GetCollection("deviceData")
-			dataSetCollection = store.GetCollection("deviceDataSets")
-			summaryCollection = store.GetCollection("summary")
-			alertsCollection = store.GetCollection("alerts")
-			Expect(store.EnsureIndexes()).To(Succeed())
+			collectionsOnce.Do(func() {
+				collection = store.GetCollection("deviceData")
+				dataSetCollection = store.GetCollection("deviceDataSets")
+				summaryCollection = store.GetCollection("summary")
+				alertsCollection = store.GetCollection("alerts")
+			})
 		})
 
 		AfterEach(func() {
-			if collection != nil {
-				collection.Database().Drop(context.Background())
-				dataSetCollection.Database().Drop(context.Background())
-				summaryCollection.Database().Drop(context.Background())
-				alertsCollection.Database().Drop(context.Background())
-			}
+			var err error
+			ctx := context.Background()
+			all := bson.D{}
+			_, err = collection.DeleteMany(ctx, all)
+			Expect(err).To(Succeed())
```

Comment:
[nit] I think you can rewrite this as:
```suggestion
            Expect(collection.DeleteMany(ctx, all)).Error().ShouldNot(HaveOccurred())
```

Reply:
And the ones below.